sdks/uniswapx-sdk/src/order/PriorityOrder.ts

```diff
@@ -0,0 +1,376 @@
+      permit2Address: this.permit2Address,
+      reactor: this.info.reactor,
+      swapper: this.info.swapper,
+      nonce: this.info.nonce.toString(),
+      deadline: this.info.deadline,
+      additionalValidationContract: this.info.additionalValidationContract,
+      additionalValidationData: this.info.additionalValidationData,
+      startBlock: this.info.startBlock.toString(),
+      input: {
+        token: this.info.input.token,
+        amount: this.info.input.amount.toString(),
+        mpsPerPriorityFeeWei: this.info.input.mpsPerPriorityFeeWei.toString(),
+      },
+      outputs: this.info.outputs.map((output) => ({
+        token: output.token,
+        amount: output.amount.toString(),
+        mpsPerPriorityFeeWei: output.mpsPerPriorityFeeWei.toString(),
+        recipient: output.recipient,
+      })),
+    };
+  }
+
+  /**
+   * @inheritdoc order
+   */
+  serialize(): string {
+    const abiCoder = new ethers.utils.AbiCoder();
+    return abiCoder.encode(PRIORITY_ORDER_ABI, [
+      [
+        [
+          this.info.reactor,
+          this.info.swapper,
+          this.info.nonce,
+          this.info.deadline,
+          this.info.additionalValidationContract,
+          this.info.additionalValidationData,
+        ],
+        this.info.startBlock,
+        [
+          this.info.input.token,
+          this.info.input.amount,
+          this.info.input.mpsPerPriorityFeeWei,
+        ],
+        this.info.outputs.map((output) => [
+          output.token,
+          output.amount,
+          output.mpsPerPriorityFeeWei,
+          output.recipient,
+        ]),
+      ],
+    ]);
+  }
+
+  /**
+   * @inheritdoc Order
+   */
+  getSigner(signature: SignatureLike): string {
+    return ethers.utils.computeAddress(
+      ethers.utils.recoverPublicKey(
+        SignatureTransfer.hash(
+          this.toPermit(),
+          this.permit2Address,
+          this.chainId,
+          this.witness()
+        ),
+        signature
+      )
+    );
+  }
+
+  /**
+   * @inheritdoc Order
+   */
+  permitData(): PermitTransferFromData {
+    return SignatureTransfer.getPermitData(
+      this.toPermit(),
+      this.permit2Address,
+      this.chainId,
+      this.witness()
+    ) as PermitTransferFromData;
+  }
+
+  /**
+   * @inheritdoc Order
+   */
+  hash(): string {
+    return ethers.utils._TypedDataEncoder
+      .from(PRIORITY_ORDER_TYPES)
+      .hash(this.witnessInfo());
+  }
+
+  /**
+   * Returns the resolved order with the given options
+   * @return The resolved order
+   */
+  resolve(options: PriorityOrderResolutionOptions): ResolvedUniswapXOrder {
+    if (options.currentBlock && options.currentBlock.lt(this.info.startBlock)) {
+      throw new OrderNotFillable("Start block in the future");
+    }
+    return {
+      input: {
+        token: this.info.input.token,
+        amount: scaleInput(this.info.input, options.priorityFee),
+      },
+      outputs: scaleOutputs(this.info.outputs, options.priorityFee),
```

Comment:
PriorityFee is something we set - this is how many mps (1e^-7 of a %) we will scale the input/output by per every wei of priority fee. In reality we will likely set this value to 1

Reply:
> how many mps (1e^-7 of a %) we will scale the input/output by per every wei of priority fee

I thought that was `mpsPerPriorityFeeWei`?

Reply:
oh sorry, you're right. This is meant to be used for off chain resolving of the order - idea is that fillers would call this func with a priorityFee to quote orders off chain

Reply:
I see, thanks! I'm still curious about the first question:
> Wouldn't we only scale either the input or the output (not both) depending on EXACT_IN vs EXACT_OUT?

Reply:
Yes that is true, and anticipated behavior is that only the input has mpsPerPriorityFeeWei or only the outputs have mpsPerPriorityFeeWei. We should throw if both have a nonzero mpsPerPriorityFeeWei in the SDK since we do within the contract as well.

Reply:
yep, there is an invariant check in the order builder exactly for this:
```
    invariant(
      !(
        this.info.input.mpsPerPriorityFeeWei.gt(0) &&
        this.info.outputs.some((output) => output.mpsPerPriorityFeeWei.gt(0))
      ),
      "Can only configure priority auction on either input or output"
    );
```